pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -460,8 +482,8 @@ func (n *NetworkPolicyController) createAppliedToGroup(npNsName string, pSel, nS
 // NetworkPolicyPeer object in NetworkPolicyRule. This function simply
 // creates the object without actually populating the PodAddresses as the
 // affected GroupMembers are calculated during sync process.
-func (n *NetworkPolicyController) createAddressGroup(namespace string, podSelector, nsSelector, eeSelector *metav1.LabelSelector) string {
-	groupSelector := antreatypes.NewGroupSelector(namespace, podSelector, nsSelector, eeSelector)
+func (n *NetworkPolicyController) createAddressGroup(namespace string, podSelector, nsSelector, eeSelector, nodeSelector *metav1.LabelSelector) string {
+	groupSelector := antreatypes.NewGroupSelector(namespace, podSelector, nsSelector, eeSelector, nodeSelector)
```

Comment:
I think if the addressGroup has nodeSelector only, we shouldn't add it to groupingInterface to avoid unncessary overhead, as we never consume its result. Otherwise it would still be processes when matching any Pods.

Reply:
Updated.